nhs/data/join_on_sa1.py

```diff
@@ -0,0 +1,219 @@
+from typing import Any, List  # Importing required types
+
+import fiona
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import numpy as np
+import polars as pl
+import yaml
+from fiona.drvsupport import supported_drivers
+from loguru import logger
+from pathos.multiprocessing import ProcessingPool as Pool
+from pyproj import CRS
+
+# Set the SHAPE_RESTORE_SHX option
+supported_drivers["ESRI Shapefile"] = "rw"
+
+
+def read_config(config_path: str) -> dict:
+    """Read YAML configuration file."""
+    with open(config_path, "r") as file:
+        config = yaml.safe_load(file)
+    logger.info(f"Configuration loaded: {config}")
+    return config
```

Comment:
Hm, maybe move this to `nhs/config.py`, we can have other config functions as well like `generate_default_config_file()`

Reply:
OK I think I've done it, not entirely sure if I did it right though

Reply:
looks good to me!